common/src/main/java/net/adoptopenjdk/icedteaweb/jvm/JvmUtils.java

```diff
@@ -54,71 +72,188 @@ private static boolean isValidStartingArgument(final String argument) {
+                "-XX:CMSInitiatingOccupancyFraction",   /* Sets the threshold percentage of the used heap */
+                "-XX:+UseCompressedOops",               /* Enables compressed references in 64-bit JVMs */
+                "-XX:GCPauseIntervalMillis",
+                "-XX:MaxGCPauseMillis",                 /* A hint to the virtual machine to pause times */
+                "-XX:+CMSIncrementalMode",              /* Enables the incremental mode */
+                "-XX:MaxMetaspaceSize",                 /* Sets an upper limit on memory used for class metadata */
+                "-XX:StringDeduplicationAgeThreshold",
+                "-XX:GCTimeLimit",
+                "-XX:GCHeapFreeLimit"
+        };
+    }
+
+    /**
+     * https://docs.oracle.com/javase/8/docs/technotes/guides/javaws/developersguide/syntax.html#secure-property
+     * <p>
+     * Properties set in the JNLP file will normally be set by Web Start after the VM is started but before
+     * the application is invoked. These properties are considered "secure" properties and can be passed as
+     * -Dkey=value arguments on the java invocation command line. The following properties are predefined secure
+     * properties and will be passed to the VM in this way.
+     *
+     * @return
+     */
+    private static String[] getValidSecureProperties() {
+        return new String[]{
+                "sun.java2d.noddraw",
+                "javaws.cfg.jauthenticator",
+                "swing.useSystemFontSettings",
+                "swing.metalTheme",
+                "http.agent",
+                "http.keepAlive",
+                "sun.awt.noerasebackground",
+                "sun.java2d.opengl",
+                "sun.java2d.d3d",
+                "java.awt.syncLWRequests",
+                "java.awt.Window.locationByPlatform",
+                "sun.awt.erasebackgroundonresize",
+                "sun.awt.keepWorkingSetOnMinimize",
+                "swing.noxp",
+                "swing.boldMetal",
+                "awt.useSystemAAFontSettings",
+                "sun.java2d.dpiaware",
+                "sun.awt.disableMixing",
+                "sun.lang.ClassLoader.allowArraySyntax",
+                "java.awt.smartInvalidate",
+                "apple.laf.useScreenMenuBar",
+                "java.net.preferIPv4Stack",
+                "java.util.Arrays.useLegacyMergeSort",
+                "sun.locale.formatasdefault",
+                "sun.awt.enableExtraMouseButtons",
+                "com.sun.management.jmxremote.local.only",
+                "sun.nio.ch.bugLevel",
+                "sun.nio.ch.disableSystemWideOverlappingFileLockCheck",
+                "jdk.map.althashing.threshold",
+                "javaplugin.lifecycle.cachesize",
+                "http.maxRedirects",
+                "http.auth.digest.validateProxy",
+                "http.auth.digest.validateServer"
+        };
+    }
+
+    public static List<String> getValidSecurePropertiesList() {
```

Comment:
this method is never used. Please delete it.

Reply:
Method may be needed in OWS.